scalableminds/webknossos
# Choose mag when training models on multiple annotations

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Enable jobs
- Open an annotation and go to AI Analysis > Train a model
- Select a ground truth and a color layer
- Select a mag. Check that all available mags are present in both layers
- Start a training / check network tab that the right mag was sent in the request
- Do the same thing, but for multiple annotations. Do that by navigating to Administration > AI models > Train new model.


- ### TODOs:
- [x] update mags after any of the input layers change 
    - [x] deselect mag
    - [x] move value to upper component
- [x] find union of data and ground truth layer
    - [x] datastructure: save map annotation id -> datalayermags, groundtruthlayermags
- [x] check that it works from within annotation and with multiple annotations
- [x] fix spacing (required vs not required fields in form, see pic)
- [x] address first review
- [ ] regarding testing: does this need to be tested after the request was sent? if yes, how?
- [x] when opening the modal, there is a ground truth layer option for the active segmentation layer, and for the fallback layer. the first one's name is simply the tracing id, which makes it harder to understand what is happening. 
![image](https://github.com/user-attachments/assets/774b507a-62fe-4aae-a3ed-3712a77b361c)



### Issues:
- contributes to #8097

------
(Please delete unneeded items, merge only when none are left open)
